lumen/ai/controls.py

```diff
@@ -49,6 +54,14 @@ def add_name(event):
             name.visible = True
         add.visible = True
 
+    def select_sheet(event):
+        if not upload.filename.endswith(".xlsx"):
+            return
+        sheets = pd.read_excel(wrap_upload(), sheet_name=None)
+        sheet_names = list(sheets.keys())
+        input_sheet.options = sheet_names
+        input_sheet.visible = True
+
     name = pn.widgets.TextInput(name="Name your table", visible=False)
     upload = pn.widgets.FileInput(align="end", accept=".csv,.parquet,.parq,.json,.xlsx")
```

Comment:
Maybe switch to FileDropper?

Reply:
Okay implemented in https://github.com/holoviz/lumen/pull/654